packages/worker/src/worker.ts

```diff
@@ -1217,6 +1218,7 @@ export class Worker {
                   try {
                     const decodedActivation = await this.workflowCodecRunner.decodeActivation(activation);
                     const unencodedCompletion = await state.workflow.activate(decodedActivation);
+                    this.validateCompletion(unencodedCompletion);
```

Comment:
Would it be clearer/possible to validate this where the local activity is invoked? You I suppose this is preferred over passing a collection of activity names into the sandbox?

Reply:
I can't pass in activity names if we want to support dynamic activities.

Reply:
That would have been my preference though.

Reply:
To make this error catchable in the workflow.

Reply:
> I can't pass in activity names if we want to support dynamic activities.

If there is a dynamic activity on the worker, instead of activity names, you tell the workflow that scheduling a local activity will never fail (i.e. that "all" names are valid).